### PR TITLE
Audio devices screen fixes and several cleanups in audio.cc and image.hh

### PR DESCRIPTION
**Pull request by Baklap4**
### What does this PR do?

Fixes the audio devices screen
and fixes some minor issues in audio.cc and image.hh

### Closes Issue(s)

Partly solves an issue from #632

### Motivation

Screen Audio devices weren't properly working in windows.

### Additional Notes

Double check if linux/mac is still alright

### CODE REVIEW

**Comment by Baklap4** [comment]
Ubuntu 21.04 still works fine

**Comment by ooshlablu** [comment]
Tested configuring a couple of devices in the audio menu, didn't notice any differences.
Play tested a couple of songs with guitar and vocals, everything works fine.

Tested on Ubuntu 20.04

**Review by ooshlablu — APPROVED** [review]
I can't review the code, but everything works fine in testing.

File: game/audio.cc [hunk 1]
```diff
@@ -151,9 +151,9 @@ bool Music::operator()(float* begin, float* end) {
 // 			if (t.pitchFactor != 0) { // Pitch shift
 // 				Buffer tempbuf(end - begin);
 // 				// Get audio to temp buffer
-// 				if (t.mpeg.audioQueue(&*tempbuf.begin(), &*tempbuf.end(), m_pos, t.fadeLevel)) eof = false;
+// 				if (t.mpeg.audioQueue(tempbuf.data(), tempbuf.data() + tempbuf.size(), m_pos, t.fadeLevel)) eof = false;
```
Comment (OznOg) [inline game/audio.cc:154]: this is commented code, isn't it?

**Review by OznOg — APPROVED** [review]
LGTM

File: game/audio.cc (hunk 1 above)
Comment (Baklap4) [reply game/audio.cc:154]: Correct. Just did a refactor it also took in account these lines it seems. Should work nontheless

**Pull request merged**